tasks/sync-granule/tests/sync_granule_test.js

```diff
@@ -294,6 +294,84 @@ test.serial('download Granule from SFTP endpoint', async (t) => {
   );
 });
 
+test.serial('download granule from S3 provider with checksum and data file in an alternate bucket', async (t) => {
+  const granuleFilePath = randomString();
+  const granuleFileName = t.context.event.input.granules[0].files[0].name;
+  const alternateDataBucket = randomString();
+  const alternateBucket = randomString();
+  const checksumFile = granuleFileName.replace('hdf', 'cksum');
+
+  t.context.event.config.provider = {
+    id: 'MODAPS',
+    protocol: 's3',
+    host: randomString(),
+  };
+
+  t.context.event.input.granules[0].files[0] = {
+    path: granuleFilePath,
+    name: granuleFileName,
+    type: 'data',
+    source_bucket: alternateDataBucket,
+  };
+  t.context.event.input.granules[0].files[1] = {
+    path: granuleFilePath,
+    name: checksumFile,
+    source_bucket: alternateBucket,
+  };
+  t.context.event.config.collection.files.push({
+    regex: '^MOD09GQ\\.A[\\d]{7}\\.[\\S]{6}\\.006\\.[\\d]{13}\\.cksum$',
+    bucket: 'protected',
+    sampleFileName: 'MOD09GQ.A2017025.h21v00.006.2017034065104.cksum',
+    url_path: 'file-example/',
+    checksumFor: '^MOD09GQ\\.A[\\d]{7}\\.[\\S]{6}\\.006\\.[\\d]{13}\\.hdf$',
+  });
+
+  await validateConfig(t, t.context.event.config);
+  await validateInput(t, t.context.event.input);
+
+  await s3().createBucket({ Bucket: alternateDataBucket }).promise();
+  t.teardown(async () => {
+    await recursivelyDeleteS3Bucket(alternateDataBucket);
+  });
+  await s3().createBucket({ Bucket: alternateBucket }).promise();
+  t.teardown(async () => {
+    await recursivelyDeleteS3Bucket(alternateBucket);
+  });
```

Comment:
🐜 I think you can combine these teardowns into one

Reply:
Fair enough.  In larger functions it may make sense to declare the teardowns following the logic they're cleaning up, ignoring the ordering issue mentioned in the docs, but....   Sure, that's probably a better pattern in this sort of case.     Will update.

Reply:
006bb605b